source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -229,7 +229,7 @@ absl::Status ProcessorState::handleBodyResponse(const BodyResponse& response) {
           const auto mut_status = MutationUtils::applyHeaderMutations(
               common_response.header_mutation(), *headers_,
               common_response.status() == CommonResponse::CONTINUE_AND_REPLACE,
-              filter_.config().mutationChecker(), filter_.stats().rejected_header_mutations_);
+              filter_.config().mutationChecker(), filter_.stats().rejected_header_mutations_, body_mode_ == ProcessingMode::STREAMED);
```

Comment:
here the last parameter can be changed into "true"

And below line 245 "headers_->removeContentLength();" can be simply removed.